source/slang/slang-check-decl.cpp

```diff
@@ -7566,8 +7649,11 @@ namespace Slang
             for (auto& m : structDecl->members)
             {
                 auto varDeclBase = as<VarDeclBase>(m);
+
+                // Static variables are initialized at start of runtime, not inside a constructor
                 if (!varDeclBase
-                    || !varDeclBase->initExpr)
+                    || !varDeclBase->initExpr
+                    || varDeclBase->hasModifier<HLSLStaticModifier>())
```

Comment:
It looks like this covers static member variables.
But I am not seeing a code that covers static variables declared inside of functions like,
```
void MyFunc()
{
    static int i;
}
```
Is it covered elsewhere? If not, you may need to.